video/out/w32_common.c

```diff
@@ -1808,8 +1838,20 @@ static int gui_thread_control(struct vo_w32_state *w32, int request, void *arg)
         s[1] *= w32->dpi_scale;
 
         RECT *rc = w32->current_fs ? &w32->prev_windowrc : &w32->windowrc;
-        const int x = rc->left + rect_w(*rc) / 2 - s[0] / 2;
-        const int y = rc->top + rect_h(*rc) / 2 - s[1] / 2;
+
+        int x, y;
+        if (w32->snapped_left)
+            x = rc->left;
+        else if (w32->snapped_right)
+            x = rc->right - s[0];
+        else
+            x = rc->left + rect_w(*rc) / 2 - s[0] / 2;
+        if (w32->snapped_top)
+            y = rc->top;
+        else if (w32->snapped_bottom)
+            y = rc->bottom - s[1];
+        else
+            y = rc->top + rect_h(*rc) / 2 - s[1] / 2;
```

Comment:
This looks like exactly the same logic, might want to enclose it in function to not duplicate it, to avoid desync in future.